### PR TITLE
fix(collapse): the items type should be optional

### PR DESCRIPTION
**Pull request by Dunqing**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

Borken in #42545

### 💡 Background and solution

The `Children` usage is still available, There should not be this breaking change

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Fix Collapse's items type should be optional       |
| 🇨🇳 Chinese |    修复 Collapse 组件 items 类型应该为可选     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 545c9d4</samp>

Refactored `CollapseProps` interface to use `Pick` utility type from `rc-collapse` library. This simplifies the code and avoids repeating the `items` property definition.

### <samp>🤖 Generated by Copilot at 545c9d4</samp>

*  Simplify `CollapseProps` interface by extending from `RcCollapseProps` ([link](https://github.com/ant-design/ant-design/pull/42877/files?diff=unified&w=0#diff-3541251bb31318316b91e1f6710b646bd92edaff4318ca790c6401bea6962fc9L23-R23))

### CODE REVIEW

**Review by MadCcc — APPROVED** [review]

**Review by kiner-tang — APPROVED** [review]

**Review by afc163 — APPROVED** [review]

**Comment by afc163** [comment]
为啥我们的 Collapse.Panel 写法的 demo ts 编译时没报错？

File: components/collapse/Collapse.tsx [hunk 1]
```diff
@@ -20,8 +20,7 @@ import useStyle from './style';
 type ExpandIconPositionLegacy = 'left' | 'right';
 export type ExpandIconPosition = 'start' | 'end' | ExpandIconPositionLegacy | undefined;
 
-export interface CollapseProps {
-  items: RcCollapseProps['items'];
+export interface CollapseProps extends Pick<RcCollapseProps, 'items'> {
```
Comment (yoyo837) [inline components/collapse/Collapse.tsx:23]: 为啥是Pick 而不是 ? optional?

Comment (Dunqing) [reply components/collapse/Collapse.tsx:23]: 和 RcCollapseProps 保持一致，避免 RC 改了 items 类型，这里没改

**Comment by Dunqing** [comment]
> 为啥我们的 Collapse.Panel 写法的 demo ts 编译时没报错？

image🤣

**Comment by afc163** [comment]
在 `__tests__/type.test.ts` 里补一个老的 Collapse Panel 写法。

File: components/collapse/__tests__/index.test.tsx [hunk 2]
```diff
@@ -209,4 +209,18 @@ describe('Collapse', () => {
...
+        <Collapse.Panel key="test panel1" header="test panel1">
+          test1
+        </Collapse.Panel>
+        <Collapse.Panel key="test panel2" header="test panel2">
+          test2
+        </Collapse.Panel>
+      </Collapse>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+  });
```
Comment (afc163) [inline components/collapse/__tests__/index.test.tsx:225]: 加个 https://github.com/ant-design/ant-design/blob/master/components/cascader/__tests__/type.test.tsx 文件吧，这样知道是专门做 ts 定义检测的。

Comment (Dunqing) [reply components/collapse/__tests__/index.test.tsx:225]: 可以等 vitest 合了做？vitest 有专门类型检查的 api

**Pull request merged**